grpc/src/main/java/com/linecorp/armeria/server/grpc/DeferredListener.java

```diff
@@ -34,8 +35,8 @@
 import io.netty.util.concurrent.EventExecutor;
 
 final class DeferredListener<I> extends ServerCall.Listener<I> {
-
     private static final List<?> NOOP_TASKS = ImmutableList.of();
+    private static final List<?> TEMPORARY_TASKS = Collections.unmodifiableList(new ArrayList<>());
```

Comment:
I find the way we handle the state is confusing. How about:

- Making `this.pendingTasks` nullable
- Adding a boolean member field that tells whether it's in the loop or not.

Then, we could do the following:

- `inLoop == false` and `this.pendingTasks == null`: no pending tasks
- `inLoop == false` and `this.pendingTasks != null`: some pending tasks. 
  - Assign `null` to `this.pendingTasks` and `true` to `inLoop` before iterating.
- While in the loop (`inLoop == true`)
  - `this.pendingTasks == null`: Assign a new ArrayList that contains the first new task to `this.pendingTasks`.
  - `this.pendingTasks != null`: Add the task to `this.pendingTasks`.
  - Assign `false` to `inLoop` after the loop is over.

Reply:
Well, Let me give it a shot

Reply:
Applied~